### DISCUSSION TITLE
No ".slc" file in the "sentinelStack.py" output from ISCE2, but it is required by Dolphin to run

### DISCUSSION THREAD

**Discussion by Shima-shoki** [isce-framework/dolphin#505] [Q&A] [answered]
Hello, I'm now testing the Dolphin to handle the ISCE2's sentinelStack.py output. This could be the problem in my ISCE2's configuration or processing problem rather than the Dolphin, but I wanted to ensure that.

 In the "Getting started" page, the list of coregistered SLC files shall be given by `dolphin config --slc-files merged/SLC/*/*.slc` following the[ ISCE2 stack processor output](https://github.com/isce-framework/isce2/tree/main/contrib/stack). As a practice, I chose 6 SLC files and ran the sentinelStack.py. The processing seemed okay after running through run_{number}_XXXX files until the end (run_13_grid_baseline), and there are list of files under merged/SLC in the output directory. However, there is no "{date}.slc" file which shall be read by Dolphin to process. I have **{date}.slc.full, {date}.slc.full.aux.xml, {date}.slc.full.vrt, {date}.slc.full.xml, and {date}.slc.hdr, but no {date}.slc**. 

![image](https://github.com/user-attachments/assets/680a72c9-2cba-47e1-b79e-f31fb98afbfc)

The {date}.slc.full data is more than a gigabite size, so I believe there is something inside it, but it looks like the file extension is not exactly the same what Dolphin requires to process. Are there something wrong with my ISCE2 software outputs, or Dolphin can also handle the .slc.full for processing?

**Comment by scottstanie** [comment, marked as answer]
Hi @Shima-shoki , see https://github.com/isce-framework/dolphin/discussions/277 : the isce2 naming of coregistered SLC files is somewhat arbitrary. The relevant part is if GDAL can recognize and read the files- in this case, there is a ".hdr" that's trying to be an ENVI file header, but there is also the `.slc.full.vrt` added specifically for GDAL. you can point it at those if the `.slc` files are not getting recognized.
You can check whether it will work by doing `gdalinfo <filename you are trying to use>`

**Comment by Shima-shoki** [comment]
@scottstanie 
Thank you very much! I should have read the discussion first before asking in the issue, sorry about that. After passing a list of .vrt files, the software seems now working.

**Reply by pyyff** [reply]
Hello, could you please explain in detail how to proceed? I have encountered a similar issue and would be very grateful.

**Reply by Shima-shoki** [reply]
@pyyff 
Hi, I specified the .vrt files instead of .slc files at the dolfin config options like below! 
Hope this helps :)

```
!dolphin config --slc-files /content/drive/MyDrive/SatelliteImageProcessings/InSAR_Python/ISCE2-GPU_SLCstack/merged/SLC/*/*.vrt --strides 20 5 --threads-per-worker 8  --n-parallel-bursts 2 --enable-gpu
```

**Reply by pyyff** [reply]
Thank you very much for your help, I was able to run it successfully and get the result, but at the beginning I got a small error, '.slc.full.vrt is   not a CSLC file', I don't know if it's an interaction issue between ISCE and Dolphin.

**Reply by scottstanie** [reply]
https://github.com/opera-adt/opera-utils/blob/main/src/opera_utils/_cslc.py do you mean you saw this warning message? Or there was an error that stopped processing?
